### PR TITLE
[language] Simple tool to print a genesis blob

### PR DESCRIPTION
**Pull request by dariorussi**
This tool takes a genesis blob as saved from the genesis generation tool
(and so a genesis transaction) and prints the content in an "inspectable" form.
The help should give enough information on what the options are

### CODE REVIEW

**Comment by dariorussi** [comment]
in case it helps here is the output of --help

Tool to display the content of a genesis blob (genesis transaction).

Takes a genesis blob transaction that can be produced by the genesis generator tool. It prints the genesis data in
different forms that can be used to review a genesis blob or to compare different genesis transactions.

The default or `--all` flag produces a somewhat pretty printing in a sorted order which can be used to review or compare
with a common `diff` command different genesis transactions. However, just performing a diff with keys (sorted and not)
can give a good sense of what are the changes in different genesis transactions.

Printing can be done by the different types of data in a genesis blob. As far as this tool is concerned the different
data types are: Modules, Resources and Events.

USAGE:
    genesis-viewer [FLAGS] --genesis-file <genesis-file>

FLAGS:
    -a, --all               
            [default] Print everything in the most verbose form and sorted, ignore all other options

        --event-keys        
            Print events key and events type

        --event-raw         
            Print raw events

    -h, --help              
            Prints help information

    -e, --type-events       
            Print events only (pretty format)

    -m, --type-modules      
            Print modules only (pretty format)

    -r, --type-resources    
            Print resources only (pretty format)

    -V, --version           
            Prints version information

    -w, --write-set         
            Print raw WriteSet only (no events)

    -t, --ws-by-type        
            Print WriteSet by type (Module/Resource, pretty format)

    -k, --ws-keys           
            Print WriteSet keys in the order defined in the binary blob

    -s, --ws-sorted-keys    
            Print WriteSet keys sorted


OPTIONS:
    -f, --genesis-file <genesis-file>

**Comment by dariorussi** [comment]
this allowed me to get a good sense of the differences (well in my case no difference but just different order) between 2 genesis blobs

**Comment by dariorussi** [comment]
in time this has to be spilt in a library that can print  from a blob `&[u8]` and a main that is what this is.
Also it should be hooked up with our disassembler for better printing of code

File: language/tools/genesis-viewer/src/main.rs [hunk 1]
```diff
@@ -0,0 +1,214 @@
...
+    let mut f = File::open(args.genesis_file.clone())
+        .unwrap_or_else(|_| panic!("Cannot open file {:?}", args.genesis_file));
+    let mut bytes = vec![];
+    f.read_to_end(&mut bytes)
+        .expect("genesis file cannot be read");
+    let txn =
+        lcs::from_bytes(&bytes).expect("genesis blob did not deserialize correctly (lcs error)");
+    if let Transaction::UserTransaction(txn) = txn {
+        if let TransactionPayload::WriteSet(ws) = txn.payload() {
+            if args.all
```
[the remark this answers is not in the document]
Comment (dariorussi) [reply language/tools/genesis-viewer/src/main.rs]: I am looking into this and looked before and cannot find an easy way to express that.
There is a default value per argument but that is not what we want here. What we are after is that if no argument is provided then `all` is the default. 
I should look into whether it is possible to check the number of args overall and maybe that is the answer?
if you have some idea please let me know

[the remark this answers is not in the document]
Comment (dariorussi) [reply language/tools/genesis-viewer/src/main.rs]: changed to use arg count which is a better way than or'ing all args value though still a bit of a hack

**Comment by opsguy** [comment]
WOOOOOOOOOOOOOOOWWWWWWWWWWWWW

THIS IS SO COOL

**Pull request closed without merging**